tests/compile-fail/data_race/release_seq_race.rs

```diff
@@ -0,0 +1,47 @@
+// ignore-windows: Concurrency on Windows is not supported yet.
+// compile-flags: -Zmiri-disable-isolation
+
+use std::thread::{spawn, sleep};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::time::Duration;
+
+#[derive(Copy, Clone)]
+struct EvilSend<T>(pub T);
+
+unsafe impl<T> Send for EvilSend<T> {}
+unsafe impl<T> Sync for EvilSend<T> {}
+
+static SYNC: AtomicUsize = AtomicUsize::new(0);
+
+pub fn main() {
+    let mut a = 0u32;
+    let b = &mut a as *mut u32;
+    let c = EvilSend(b);
+
+    unsafe {
+        let j1 = spawn(move || {
+            *c.0 = 1;
+            SYNC.store(1, Ordering::Release);
+            sleep(Duration::from_millis(100));
+            SYNC.store(3, Ordering::Relaxed);
+        });
+
+        let j2 = spawn(move || {
+            // Blocks the acquire-release sequence
+            SYNC.store(2, Ordering::Relaxed);
+        });
+
+        let j3 = spawn(move || {
+            sleep(Duration::from_millis(1000));
+            if SYNC.load(Ordering::Acquire) == 3 {
+                *c.0 //~ ERROR Data race
```

Comment:
Oh this is a nasty test, I like it. :D

Same comments wrt. scheduler-dependence as before applies, though.